Could not find a declaration file for module 'eslint'

## Current behavior
My Webpack config file is written in TS and type-checked on test/release.

With v4 of `fork-ts-checker-webpack-plugin` **tsc** started to complain about a not found declaration file for `eslint`

```
node_modules/fork-ts-checker-webpack-plugin/lib/issue/eslint/EsLintIssueFactory.d.ts:1:25 - error TS7016: Could not find a declaration file for module 'eslint'. '/MY_PROJECT_DIR/node_modules/eslint/lib/api.js' implicitly has an 'any' type.
  Try `npm install @types/eslint` if it exists or add a new declaration (.d.ts) file containing `declare module 'eslint';`

1 import * as eslint from 'eslint';
                          ~~~~~~~~

node_modules/fork-ts-checker-webpack-plugin/lib/issue/eslint/FileAwareEsLintMessage.d.ts:1:25 - error TS7016: Could not find a declaration file for module 'eslint'. '/MY_PROJECT_DIR/node_modules/eslint/lib/api.js' implicitly has an 'any' type.
  Try `npm install @types/eslint` if it exists or add a new declaration (.d.ts) file containing `declare module 'eslint';`

1 import * as eslint from 'eslint';
```

## Expected behavior
type-checking should pass

## Steps to reproduce the issue
**tsconfig.json**
```jsonc
{
  "compilerOptions": {
    "noEmit": true,
    "target": "es5",
    "module": "commonjs",
    "esModuleInterop": true,
    "lib": ["es6", "dom", "es2016.array.include"],
    "importHelpers": true,
    "sourceMap": true,
    "outDir": "./dist",
    "jsx": "preserve",
    "alwaysStrict": true,
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "forceConsistentCasingInFileNames": true,
    "types": ["node", "jest", "jest-enzyme"],
    "typeRoots": ["./typings", "./node_modules/@types"],
    "baseUrl": ".",
    "paths": {
      "*": ["*", "typings/*"]
    }
  },
  "include": [
    "./src",
    "./scripts",
    "./config",
    "./webpack.config.ts",
    "./typings"
  ]
}
```
**webpack.config.ts**
```ts
import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';

export = {
  // ... various webpack configuration

 plugins: [
  new ForkTsCheckerWebpackPlugin()
 ]
}
```
and then run

```sh
npx tsc
```

## Environment
 - **fork-ts-checker-webpack-plugin**: 4.0.1
 - **typescript**: 3.7.5
 - **~~tslint~~**
 - **eslint**: 6.8.0
 - **webpack**: 4.41.5
 - **os**: Ubuntu 18.04

--- a/package.json
+++ b/package.json
@@ -85,7 +85,6 @@
   "devDependencies": {
     "@commitlint/config-conventional": "^7.5.0",
     "@types/babel-code-frame": "^6.20.1",
-    "@types/eslint": "^4.16.6",
     "@types/jest": "^24.0.11",
     "@types/lodash": "^4.14.134",
     "@types/micromatch": "^3.1.0",

--- a/src/ApiIncrementalChecker.ts
+++ b/src/ApiIncrementalChecker.ts
@@ -1,5 +1,3 @@
-import * as eslint from 'eslint'; // Imported for types alone
-
 import {
   IncrementalCheckerInterface,
   IncrementalCheckerParams
@@ -11,11 +9,12 @@ import {
   createIssuesFromTsDiagnostics,
   createIssuesFromEsLintReports
 } from './issue';
+import { LintReport } from './types/eslint';
 
 export class ApiIncrementalChecker implements IncrementalCheckerInterface {
   protected readonly tsIncrementalCompiler: CompilerHost;
 
-  private currentEsLintErrors = new Map<string, eslint.CLIEngine.LintReport>();
+  private currentEsLintErrors = new Map<string, LintReport>();
   private lastUpdatedFiles: string[] = [];
   private lastRemovedFiles: string[] = [];
 

--- a/src/FilesRegister.ts
+++ b/src/FilesRegister.ts
@@ -1,10 +1,10 @@
 import * as ts from 'typescript'; // import for types alone
-import * as eslint from 'eslint'; // import for types alone
+import { LintReport } from './types/eslint';
 
 export interface DataShape {
   source?: ts.SourceFile;
   linted: boolean;
-  eslints: eslint.CLIEngine.LintReport[];
+  eslints: LintReport[];
 }
 
 export class FilesRegister {

--- a/src/IncrementalChecker.ts
+++ b/src/IncrementalChecker.ts
@@ -1,7 +1,6 @@
 import * as fs from 'fs';
 import * as path from 'path';
 import * as ts from 'typescript'; // Imported for types alone; actual requires take place in methods below
-import * as eslint from 'eslint';
 
 import { FilesRegister } from './FilesRegister';
 import { CancellationToken } from './CancellationToken';
@@ -22,6 +21,7 @@ import {
   createIssuesFromEsLintReports,
   createIssuesFromTsDiagnostics
 } from './issue';
+import { LintReport } from './types/eslint';
 
 export class IncrementalChecker implements IncrementalCheckerInterface {
   private files = new FilesRegister(() => ({
@@ -261,7 +261,7 @@ export class IncrementalChecker implements IncrementalCheckerInterface {
           !IncrementalChecker.isFileExcluded(filePath)
       );
 
-    const currentEsLintErrors = new Map<string, eslint.CLIEngine.LintReport>();
+    const currentEsLintErrors = new Map<string, LintReport>();
     filesToLint.forEach(fileName => {
       cancellationToken.throwIfCancellationRequested();
 
@@ -289,7 +289,7 @@ export class IncrementalChecker implements IncrementalCheckerInterface {
 
     const reports = this.files
       .keys()
-      .reduce<eslint.CLIEngine.LintReport[]>(
+      .reduce<LintReport[]>(
         (innerLints, filePath) =>
           innerLints.concat(this.files.getData(filePath).eslints),
         []

--- a/src/createEslinter.ts
+++ b/src/createEslinter.ts
@@ -1,16 +1,16 @@
-import * as eslint from 'eslint'; // import for types alone
 import * as path from 'path';
 
+import { LintReport } from './types/eslint';
 import { throwIfIsInvalidSourceFileError } from './FsHelper';
 
 export function createEslinter(eslintOptions: object) {
   // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const { CLIEngine }: typeof eslint = require('eslint');
+  const { CLIEngine } = require('eslint');
 
   // See https://eslint.org/docs/1.0.0/developer-guide/nodejs-api#cliengine
   const eslinter = new CLIEngine(eslintOptions);
 
-  function getReport(filepath: string) {
+  function getReport(filepath: string): LintReport | undefined {
     try {
       if (
         eslinter.isPathIgnored(filepath) ||

--- a/src/issue/eslint/EsLintIssueFactory.ts
+++ b/src/issue/eslint/EsLintIssueFactory.ts
@@ -1,8 +1,8 @@
-import * as eslint from 'eslint';
 import { FileAwareEsLintMessage } from './FileAwareEsLintMessage';
 import { deduplicateAndSortIssues, Issue } from '../Issue';
 import { IssueOrigin } from '../IssueOrigin';
 import { IssueSeverity } from '../IssueSeverity';
+import { LintReport, LintResult } from '../../types/eslint';
 
 function createIssueFromEsLintMessage(message: FileAwareEsLintMessage): Issue {
   return {
@@ -18,7 +18,7 @@ function createIssueFromEsLintMessage(message: FileAwareEsLintMessage): Issue {
 }
 
 function createFileAwareEsLintMessagesFromEsLintResult(
-  result: eslint.CLIEngine.LintResult
+  result: LintResult
 ): FileAwareEsLintMessage[] {
   return result.messages.map(message => ({
     ...message,
@@ -27,7 +27,7 @@ function createFileAwareEsLintMessagesFromEsLintResult(
 }
 
 function createFileAwareEsLintMessagesFromEsLintReport(
-  report: eslint.CLIEngine.LintReport
+  report: LintReport
 ): FileAwareEsLintMessage[] {
   return report.results.reduce<FileAwareEsLintMessage[]>(
     (messages, result) => [
@@ -39,7 +39,7 @@ function createFileAwareEsLintMessagesFromEsLintReport(
 }
 
 function createFileAwareEsLintMessagesFromEsLintReports(
-  reports: eslint.CLIEngine.LintReport[]
+  reports: LintReport[]
 ): FileAwareEsLintMessage[] {
   return reports.reduce<FileAwareEsLintMessage[]>(
     (messages, report) => [
@@ -56,9 +56,7 @@ function createIssuesFromEsLintMessages(
   return deduplicateAndSortIssues(messages.map(createIssueFromEsLintMessage));
 }
 
-function createIssuesFromEsLintReports(
-  reports: eslint.CLIEngine.LintReport[]
-): Issue[] {
+function createIssuesFromEsLintReports(reports: LintReport[]): Issue[] {
   return createIssuesFromEsLintMessages(
     createFileAwareEsLintMessagesFromEsLintReports(reports)
   );

--- a/src/issue/eslint/FileAwareEsLintMessage.ts
+++ b/src/issue/eslint/FileAwareEsLintMessage.ts
@@ -1,10 +1,10 @@
-import * as eslint from 'eslint';
+import { LintMessage } from '../../types/eslint';
 
 /**
  * We need to define custom interface because of eslint architecture which
  * groups lint messages per file
  */
-interface FileAwareEsLintMessage extends eslint.Linter.LintMessage {
+interface FileAwareEsLintMessage extends LintMessage {
   filePath?: string;
 }
 

--- a/src/types/eslint.ts
+++ b/src/types/eslint.ts
@@ -1,0 +1,36 @@
+// copy eslint types as installation of eslint package is optional
+export interface LintMessage {
+  column: number;
+  line: number;
+  endColumn?: number;
+  endLine?: number;
+  ruleId: string | null;
+  message: string;
+  nodeType: string;
+  fatal?: true;
+  severity: 0 | 1 | 2;
+  fix?: {
+    range: [number, number];
+    text: string;
+  };
+  source: string | null;
+}
+
+export interface LintResult {
+  filePath: string;
+  messages: LintMessage[];
+  errorCount: number;
+  warningCount: number;
+  fixableErrorCount: number;
+  fixableWarningCount: number;
+  output?: string;
+  source?: string;
+}
+
+export interface LintReport {
+  results: LintResult[];
+  errorCount: number;
+  warningCount: number;
+  fixableErrorCount: number;
+  fixableWarningCount: number;
+}

--- a/test/unit/issue/eslint/EsLintIssueFactory.spec.ts
+++ b/test/unit/issue/eslint/EsLintIssueFactory.spec.ts
@@ -1,11 +1,11 @@
 import { createIssuesFromEsLintReports } from '../../../../lib/issue';
-import * as eslint from 'eslint';
+import {
+  LintMessage,
+  LintReport,
+  LintResult
+} from '../../../../lib/types/eslint';
 
 describe('[UNIT] issue/eslint/EsLintIssueFactory', () => {
-  type LintMessage = eslint.Linter.LintMessage;
-  type LintResult = eslint.CLIEngine.LintResult;
-  type LintReport = eslint.CLIEngine.LintReport;
-
   const ES_LINT_MESSAGE_ERROR: LintMessage = {
     column: 0,
     line: 13,

